src/process.cr

```diff
@@ -415,6 +415,79 @@ class Process
   private def close_io(io)
     io.close if io
   end
+
+  # Sets the real, effective, and saved `User` to the one specified.
+  def self.become(user : System::User) : Nil
+    return if LibC.setuid(user.uid) == 0
```

Comment:
There are many methods like this in the form `return if cond; raise`.
I think the control flow would be easier readable if those are written as `unless cond; raise; end` (with proper formatting of course). That would be more the Crystal way. There is no need for empty return statements.